lib/config/formatter.go

```diff
@@ -163,14 +165,75 @@ func (tf *textFormatter) Format(e *log.Entry) ([]byte, error) {
 	// write caller last if enabled
 	if tf.callerEnabled && caller != "" {
 		w.writeField(caller, noColor)
-
 	}
 
 	w.WriteByte('\n')
 	data = w.Bytes()
 	return data, nil
 }
 
+// jsonFormatter implements the logrus.Formatter interface and adds extra
+// fields to log entries
+type jsonFormatter struct {
+	log.JSONFormatter
+
+	callerEnabled    bool
+	componentEnabled bool
+}
+
+// CheckAndSetDefaults checks and sets log format configuration
+func (j *jsonFormatter) CheckAndSetDefaults(extraFields []string) error {
```

Comment:
for sure! 
personally I dislike the `CheckAndSetDefaults` function because there is a real possibility that someone's going to init the struct and not call it. I much prefer to use initializers like `NewJsonFormatter` to do both the initialization and the "check and set defaults". I was kind of aiming at that here without making too much change from what people are used to 😅